src/main.cpp

```diff
@@ -242,29 +245,56 @@ Napi::Value DDWAFContext::run(const Napi::CallbackInfo& info) {
     Napi::Error::New(env, "Calling run on a disposed context").ThrowAsJavaScriptException();
     return env.Null();
   }
-  if (info.Length() < 2) {  // inputs, timeout
-    Napi::Error::New(env, "Wrong number of arguments, expected 2").ThrowAsJavaScriptException();
+
+  if (info.Length() < 2) {  // payload, timeout
+    Napi::Error::New(env, "Wrong number of arguments, 2 expected").ThrowAsJavaScriptException();
     return env.Null();
   }
+
   if (!info[0].IsObject()) {
-    Napi::TypeError::New(env, "First argument must be an object").ThrowAsJavaScriptException();
+    Napi::TypeError::New(
+            env,
+            "Payload data must be an object")
+        .ThrowAsJavaScriptException();
     return env.Null();
   }
+
+  Napi::Value persistent = info[0].As<Napi::Object>().Get("persistent");
+  Napi::Value ephemeral = info[0].As<Napi::Object>().Get("ephemeral");
```

Comment:
Isn't a casting an expensive operation? why not:
```suggestion
  Napi::Object payload = info[0].As<Napi::Object>()
  Napi::Value persistent = payload.Get("persistent");
  Napi::Value ephemeral = payload.Get("ephemeral");
```

Reply:
If I recall correctly that will call to `napi_coerce_to_object` function which will do a `reinterpret_cast`. That type of casts don't have impact in runtime but probably the way you proposed it's more idiomatic.